### ISSUE TITLE
Virtuoso associations

### ISSUE THREAD

**Issue by justgo129** [USGCRP/gcis#378]
It seems that our version of Virtuoso is actually enabled to do some inferencing.  Run [this] (http://data.globalchange.gov/sparql?default-graph-uri=&query=select+%0D%0A+DISTINCT+%28%3Fauthor+as+%3FAuthor%29%0D%0A+%28%3Fpub+as+%3FOrganization%29%0D%0A+%0D%0ACONCAT%28str%28%3Fln%29%2C+%22%2C+%22%2C+str%28%3Fgn%29%29+as+%3FName+%0D%0A+str%28%3Forg_name%29+as+%3FOrganizationName%0D%0A+FROM+%3Chttp%3A%2F%2Fdata.globalchange.gov%3E+where+%7B+%0D%0A+%23+%3Fpub+a++UNION+%7Bgcis%3AAcademicArticle+gcis%3AReport+%7D+.%0D%0A+++%3Fpub+a++gcis%3AAcademicArticle.%0D%0A+++%3Fpub+prov%3AqualifiedAttribution+%5B+prov%3AhadRole+%3Frole+%3B+prov%3Aagent+%3Fauthor+%3B+prov%3AactedOnBehalfOf+%3Forg+%5D+.%0D%0A+++%3Fauthor+foaf%3AgivenName+%3Fgn+.%0D%0A+++%3Fauthor+foaf%3AlastName+%3Fln+.%0D%0A+++%3Forg+skos%3AprefLabel+%3Forg_name%0D%0A%7D%0D%0Aorder+by+asc%28%3Forg%29&should-sponge=&format=text%2Fhtml&timeout=0&debug=on) query and look at the myriad of organization names that are abbreviations.

In effect, Virtuoso is duplicating material in the case of orgs that have a lexicon.  There are a ton of examples here, most notably lines 139-143.  Because we have different lexicons for USDA ARS, this person is reported as working at an org called by each lexicon:  The person works at 005:18 (omb lexicon), ARS (govman lexicon), and some other names all known by another lexicon.

In short, Virtuoso is considering the value for an entity to differ by lexicon, causing it to differentiate these by lexicon, rather than by equating them through one entry.

**Comment by rewolfe** [comment]
@justgo129 - This is something that Amrutha and I saw yesterday.  This
seems to be related to the "skos:exactMatch" relationship. It would be good
if there were a way to turn off these extra return values (perhaps with an
additional clause in SPARQL).

On Tue, Jun 21, 2016 at 12:47 PM, justgo129 <notifications@github.com>
wrote:

> Assigned #378 <https://github.com/USGCRP/gcis/issues/378> to @rewolfe
> <https://github.com/rewolfe>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/USGCRP/gcis/issues/378#event-699339787>, or mute the
> thread
> <https://github.com/notifications/unsubscribe/AHR7GUg4YO23L7DX6_MZnz_wfF0vGUEQks5qOBWYgaJpZM4I68Gb>
> .
>



-- 
Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by justgo129** [comment]
Does [https://www.w3.org/TR/skos-primer/#secskoscontainment](this) help?

**Comment by rewolfe** [comment]
@justgo129 - This might be a good question for Stephan.

On Tue, Jun 21, 2016 at 2:55 PM, justgo129 <notifications@github.com> wrote:

> Does https://www.w3.org/TR/skos-primer/#secskoscontainment <http://this>
> help?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/USGCRP/gcis/issues/378#issuecomment-227537093>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AHR7GRmvLYxhq0FfaTl10hrPfwO3Ixqmks5qODOpgaJpZM4I68Gb>
> .
>



-- 
Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by justgo129** [comment]
Good call.  @zednis could you help us out with this?

**Comment by zednis** [comment]
I think I may be missing some context.

The query you linked above is the following:
```
SELECT
 DISTINCT (?author as ?Author)
 (?pub as ?Organization)
 CONCAT(str(?ln), ", ", str(?gn)) as ?Name 
 str(?org_name) as ?OrganizationName
FROM <http://data.globalchange.gov> 
WHERE { 
   ?pub a  gcis:AcademicArticle.
   ?pub prov:qualifiedAttribution [ prov:hadRole ?role ; prov:agent ?author ; prov:actedOnBehalfOf ?org ] .
   ?author foaf:givenName ?gn .
   ?author foaf:lastName ?ln .
   ?org skos:prefLabel ?org_name
}
order by asc(?org)
```

yasgui link: http://yasgui.org/short/SJYM1Sx8

example results:

| Index  | Author | Organization | Name | OrganizationName |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| 1 | http://data.globalchange.gov/person/11049 | http://data.globalchange.gov/article/10.2193/2008-265 | Edwards, Andrew J. | 1854 Treaty Authority |
| 2 | http://data.globalchange.gov/person/3005 | http://data.globalchange.gov/article/10.1007/s00382-011-1226-7 | Vermeer, Martin | Aalto University Department of Surveying |
| 3 | http://data.globalchange.gov/person/3005 | http://data.globalchange.gov/article/10.1073/pnas.1015619108 | Vermeer, Martin | Aalto University Department of Surveying |
| 4 | http://data.globalchange.gov/person/3005 | http://data.globalchange.gov/article/10.1007/s00382-011-1226-7 | Vermeer, Martin | Aalto University School of Engineering |
| 5 | http://data.globalchange.gov/person/3005 | http://data.globalchange.gov/article/10.1073/pnas.1015619108 | Vermeer, Martin | Aalto University School of Engineering |

I am not sure what duplication you are referring to in the query results and why you think this is happening because of reasoning.

PS - I did notice that you are calling the publication the ?organization variable in the query results.

**Comment by zednis** [comment]
I think I see the duplication in the following example:

| Index  | Author | Organization | Name | OrganizationName |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| 23 | http://data.globalchange.gov/person/3839 | http://data.globalchange.gov/organization/agency-toxic-substances-disease-registry | Younger, Margalit | Agency for Toxic Substances and Disease Registry |
| 24 | http://data.globalchange.gov/person/3839 | http://data.globalchange.gov/organization/agency-toxic-substances-disease-registry | Younger, Margalit | ATSDR |

**Comment by zednis** [comment]
One immediate problem is you have multiple instances of ``skos:prefLabel`` which is expressly disallowed.

from the bottom of http://data.globalchange.gov/organization/agency-toxic-substances-disease-registry.thtml:
```
## Also known as:
<http://data.globalchange.gov/organization/agency-toxic-substances-disease-registry>
   skos:prefLabel "Agency for Toxic Substances and Disease Registry";
   skos:exactMatch <http://data.globalchange.gov/lexicon/govman/agencyName/Agency%20for%20Toxic%20Substances%20and%20Disease%20Registry> .
<http://data.globalchange.gov/organization/agency-toxic-substances-disease-registry>
   skos:prefLabel "ATSDR";
   skos:exactMatch <http://data.globalchange.gov/lexicon/govman/Agency/ATSDR> .
```

The acronym label should **NOT** be a ``skos:prefLabel``.

**Comment by lomky** [comment]
Looks like we will need to update [lib/Tuba/files/templates/other_identifiers.ttl.tut](https://github.com/USGCRP/gcis/blob/7612780cb5e9e7c215e97c3b022d380c6c395111/lib/Tuba/files/templates/other_identifiers.ttl.tut). @amruelama & I are planning on working on this together as a good perlish code change to work through the development process of.

**Comment by zednis** [comment]
The issue I see here is that the ``skos:prefLabel`` is being asserted inside the for loop so ``<<%= current_resource %>>`` is getting a ``skos:prefLabel`` for each match iterated over in the loop.

```
    % for my $exterm (@$matches) {  # <!-- loop or all matches
<<%= current_resource %>>
      % if (my $url = $exterm->native_url) {
   skos:prefLabel "<%= $exterm->term %>"; # <!-- asserts a prefLabel
   gcis:hasURL "<%= $url %>";
      % } else {
   skos:prefLabel "<%= $exterm->term %>"; # <!-- asserts a prefLabel
      % }
      % if (my $same_as = $exterm->same_as) {
   owl:sameAs <<%= $same_as %>>;
      % }
   skos:exactMatch <<%= uri($exterm) %>> .
    % }
% }
```

Only 1 ``skos:prefLabel`` should be asserted for ``<<%= current_resource %>>``

**Comment by lomky** [comment]
@zednis The [base turtle for organizations](https://github.com/USGCRP/gcis/blob/7612780cb5e9e7c215e97c3b022d380c6c395111/lib/Tuba/files/templates/organization/object.ttl.tut) seems to assert the `skos:prefLabel ` of the `<<%= current_resource %>>` up at the top of the full turtle code. So my thinking is that the `other_identifiers.ttl.tut` should only assert the `skos:altLabel` you suggested. Does this seem reasonable?

**Comment by lomky** [comment]
Changes merged in pull request #396 and will go out in the next release
